huaweicloud/services/cph/resource_huaweicloud_cph_server.go

```diff
@@ -0,0 +1,802 @@
+			},
+			"vpc_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `The ID of VPC which the cloud server belongs to`,
+			},
+			"subnet_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `The ID of subnet which the cloud server belongs to`,
+			},
+			"public_ip": {
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Elem:        cphServerPublicIpSchema(),
+				Required:    true,
+				ForceNew:    true,
+				Description: `To configure an EIP for a cloud server.`,
+			},
+			"bandwidth": {
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Elem:        cphServerBandWidthSchema(),
+				Required:    true,
+				ForceNew:    true,
+				Description: `The bandwidth used by the cloud phone.`,
+			},
+			"period_unit": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `The charging period unit.`,
+				ValidateFunc: validation.StringInSlice([]string{
+					"month", "year",
+				}, false),
+			},
+			"period": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				ForceNew:     true,
+				Description:  `The charging period.`,
+				ValidateFunc: validation.IntBetween(1, 9),
+			},
+			"auto_renew": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Whether auto renew is enabled. Valid values are "true" and "false".`,
+			},
+			"availability_zone": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Computed:    true,
+				Description: `The name of the AZ where the cloud server is located.`,
+			},
+			"keypair_name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `The key pair name, which is used for logging in to the cloud phone through ADB.`,
+			},
+			"enterprise_project_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `the enterprise project ID.`,
+			},
+			"ports": {
+				Type:        schema.TypeList,
+				Elem:        cphServerApplicationPortSchema(),
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `The application port enabled by the cloud phone.`,
+			},
+			"order_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The order ID.`,
+			},
+			"addresses": {
+				Type:        schema.TypeList,
+				Elem:        cphServerAddressSchema(),
+				Computed:    true,
+				Description: `The IP addresses of the CPH server.`,
+			},
+			"status": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: `The CPH server status.`,
+			},
+		},
+	}
+}
+
+func cphServerBandWidthSchema() *schema.Resource {
+	sc := schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"bandwidth_share_type": {
```

Comment:
This parameter should **share_type** without ``bandwidth_`, and the following parameters should be updated

Reply:
ok